packages/frontend/src/components/Explorer/CustomMetricModal/FilterForm.tsx

```diff
@@ -0,0 +1,109 @@
+import {
+    ConditionalOperator,
+    createFilterRuleFromField,
+    FieldTarget,
+    FilterRule,
+    getFieldRef,
+    isDimension,
+} from '@lightdash/common';
+import { Button, Stack } from '@mantine/core';
+import { Dispatch, FC, SetStateAction, useCallback } from 'react';
+import { useExplorerContext } from '../../../providers/ExplorerProvider';
+import FilterRuleForm from '../../common/Filters/FilterRuleForm';
+import { useFiltersContext } from '../../common/Filters/FiltersProvider';
+import { addFieldRefToFilterRule } from './utils';
+
+export interface MetricFilterRuleWithFieldId
+    extends FilterRule<
+        ConditionalOperator,
+        FieldTarget & { fieldRef: string }
+    > {}
+
+export const FilterForm: FC<{
+    defaultFilterRuleFieldId: string | undefined;
+    customMetricFiltersWithIds: MetricFilterRuleWithFieldId[];
+    setCustomMetricFiltersWithIds: Dispatch<
+        SetStateAction<MetricFilterRuleWithFieldId[]>
+    >;
+}> = ({
+    defaultFilterRuleFieldId,
+    customMetricFiltersWithIds,
+    setCustomMetricFiltersWithIds,
+}) => {
+    const isEditMode = useExplorerContext(
+        (context) => context.state.isEditMode,
+    );
+    const { fieldsMap } = useFiltersContext();
+
+    const dimensions = Object.values(fieldsMap).filter(isDimension);
+
+    const addFieldRule = useCallback(() => {
+        const field = fieldsMap[defaultFilterRuleFieldId ?? 0];
```

Comment:
I don't think using an index would work in a map. 

<img width="305" alt="Captura de ecrã 2023-06-20, às 10 49 13" src="https://github.com/lightdash/lightdash/assets/9117144/d08dc191-7a43-4603-96cc-5c353983f982">

Suggestion
```
const fallbackField = Object.values(fieldsMap)[0];
const defaultField = defaultFilterRuleFieldId ? fieldsMap[defaultFilterRuleFieldId] : undefined;
const field = defaultField || fallbackField;

if(!field) {
   return;
}

```

Reply:
oops, got mixed up with the values on line 38 and that. Changing! thanks 😄